context.go

```diff
@@ -67,6 +66,92 @@ type SpanContext struct {
 	//
 	// See JaegerDebugHeader in constants.go
 	debugID string
+
+	// samplingState is shared across all spans
+	samplingState *samplingState
+}
+
+type samplingState struct {
+	flags atomic.Int32 // Only lower 8 bits are used. We use an int32 instead of a byte to use CAS operations
+}
+
+func (s *samplingState) setSampled() {
+	// TODO: add tests for all these transitions
+	swapped := s.flags.CAS(flagUnsampled, flagSampled)
```

Comment:
Interesting - I think this could work in our case because the only time we modify flags in the other direction is with `resetFlags` which doesn't involve spinning.